panel/chat/feed.py

```diff
@@ -407,13 +412,10 @@ async def _schedule_placeholder(
         if self.placeholder_threshold == 0:
             return
 
-        callable_is_async = asyncio.iscoroutinefunction(
-            self.callback
-        ) or isasyncgenfunction(self.callback)
         start = asyncio.get_event_loop().time()
         while not task.done() and num_entries == len(self._chat_log):
             duration = asyncio.get_event_loop().time() - start
-            if duration > self.placeholder_threshold or not callable_is_async:
```

Comment:
Since it's a separate thread, it's no longer needed to always show placeholder when sync.